cmd/bosun/conf/conf.go

```diff
@@ -1134,7 +1118,16 @@ func (c *Conf) AlertTemplateStrings() (*AlertTemplateStrings, error) {
 }
 
 func (c *Conf) NewExpr(s string) (*expr.Expr, error) {
-	return expr.New(s, c.Funcs())
+	ex, err := expr.New(s, c.Funcs())
+	if err == nil {
+		switch ex.Root.Return() {
+		case eparse.TypeNumber, eparse.TypeScalar:
+			break
+		default:
+			c.errorf("expression must return a number")
```

Comment:
This function should either return an error in this case, or remove the error return from its signature and instead always call c.error if it encounters one. I'd favor the latter, since that's how everything else in this package works.